Document `mask-*` utilities

danhollick opened the pull request:
Note: this is using a custom build of Tailwind that we'll need to remove before merging.

reinink commented on src/docs/mask-clip.mdx:
```diff
@@ -0,0 +1,65 @@
+    ["mask-clip-fill", "mask-clip: fill-box;"],
+    ["mask-clip-stroke", "mask-clip: stroke-box;"],
+    ["mask-clip-view", "mask-clip: view-box;"],
+    ["mask-clip-no-clip", "mask-clip: no-clip;"],
```
This has been renamed:

```suggestion
    ["mask-no-clip", "mask-clip: no-clip;"],
```

danhollick commented:
FYI, this is currently broken. I added the new `mask-position-[]`, `mask-position-()`, `mask-size-[]`, and `mask-size-()` utilities and updated anywhere I was using the old syntax. Because these aren't included in this build those places now look broken:
<img width="723" alt="Screenshot 2025-03-28 at 17 03 17" src="https://github.com/user-attachments/assets/ab375e40-dcdc-417a-a815-8ac39e1a760c" />

reinink reviewed (changes requested):
Here's my WIP feedback 😊

reinink commented on src/docs/mask-clip.mdx:
````diff
@@ -0,0 +1,65 @@
+
+```html
+<!-- [!code classes:mask-clip-border,mask-clip-padding,mask-clip-content] -->
+<div class="border-3 mask-[url(/img/circle.png)] mask-clip-border p-1.5 ..."></div>
+<div class="border-3 mask-[url(/img/circle.png)] mask-clip-padding p-1.5 ..."></div>
+<div class="border-3 mask-[url(/img/circle.png)] mask-clip-content p-1.5 ..."></div>
````
Tweak example on `mask-clip` page
